tests/relay/test_schema.py

```diff
@@ -356,3 +356,47 @@ def fruits(self) -> List[Fruit]:
             "edges": [{"node": {"id": to_base64("Fruit", i)}} for i in range(10)]
         }
     }
+
+
+def test_overwrite_resolve_id_and_no_node_id(mocker: MockerFixture):
+    mocker.patch.object(
+        DEFAULT_SCALAR_REGISTRY[relay.GlobalID],
+        "description",
+        "__GLOBAL_ID_DESC__",
+    )
+
+    @strawberry.type
+    class Fruit(relay.Node):
+        color: str
+
+        @classmethod
+        def resolve_id(cls, root) -> str:
+            return "test"  # pragma: no cover
+
+    @strawberry.type
+    class Query:
+        fruit: Fruit
+
+    expected_type = '''type Fruit implements Node {
+  """The Globally Unique ID of this object"""
+  id: GlobalID!
+  color: String!
+}
+
+"""__GLOBAL_ID_DESC__"""
+scalar GlobalID @specifiedBy(url: "https://relay.dev/graphql/objectidentification.htm")
+
+"""An object with a Globally Unique ID"""
+interface Node {
+  """The Globally Unique ID of this object"""
+  id: GlobalID!
+}
+
+type Query {
+  fruit: Fruit!
+}
+    '''
```

Comment:
nitpick to keep this using the same style as the other tests in this file

```suggestion
    expected_type = textwrap.dedent(
        '''
        type Fruit implements Node {
        """The Globally Unique ID of this object"""
        id: GlobalID!
        color: String!
      }

      """__GLOBAL_ID_DESC__"""
      scalar GlobalID @specifiedBy(url: "https://relay.dev/graphql/objectidentification.htm")

      """An object with a Globally Unique ID"""
      interface Node {
        """The Globally Unique ID of this object"""
        id: GlobalID!
      }

      type Query {
        fruit: Fruit!
      }
      '''
```